### PR TITLE
8318962: Update ProcessTools javadoc with suggestions in 8315097

### PR DESCRIPTION
**Pull request by lkorinth**
Updates to javadoc after improvement ideas in the review of 8315097. I have also removed the incomplete invocation command line. I also removed incomplete information in `executeTestJvm` and added a link to `createLimitedTestJavaProcessBuilder`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318962](https://bugs.openjdk.org/browse/JDK-8318962): Update ProcessTools javadoc with suggestions in 8315097 (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16430/head:pull/16430` \
`$ git checkout pull/16430`

Update a local copy of the PR: \
`$ git checkout pull/16430` \
`$ git pull https://git.openjdk.org/jdk.git pull/16430/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16430`

View PR using the GUI difftool: \
`$ git pr show -t 16430`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16430.diff">https://git.openjdk.org/jdk/pull/16430.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16430#issuecomment-1786623892)

### CODE REVIEW

File: test/lib/jdk/test/lib/process/ProcessTools.java [hunk 1]
```diff
@@ -566,14 +608,12 @@ public static ProcessBuilder createLimitedTestJavaProcessBuilder(String... comma
...
-     * Executes a test jvm process, waits for it to finish and returns the process output.
-     * The default jvm options from jtreg, test.vm.opts and test.java.opts, are added.
-     * The java from the test.jdk is used to execute the command.
-     * <p>
-     * The command line will be like:
-     * {test.jdk}/bin/java {test.vm.opts} {test.java.opts} cmds
-     * <p>
-     * The jvm process will have exited before this method returns.
+     * Executes a test jvm process, waits for it to finish and returns
+     * the process output.
```
Comment (RogerRiggs) [inline test/lib/jdk/test/lib/process/ProcessTools.java:612]: Dropping the support test.vm.opts and test.java.opts deserves to be explicitly documented.

Can you confirm that all existing uses no longer need the test options and have already been remediated in 8315097.

**Comment by lkorinth** [comment]
Sorry Roger, I went a bit funny in the head...

I updated the documentation to correctly state that the propagating API is used.

**Comment by RogerRiggs** [comment]
WRT the bug title and PR title...

The leading "Javadoc:" implies a change to the javadoc tool. Perhaps
```
Update ProcessTools javadoc with suggestions in 8315097
```

**Review by RogerRiggs — APPROVED** [review]

**Pull request closed without merging**